packages/strapi-plugin-email/admin/src/containers/Settings/index.js

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect } from 'react';
+import { useIntl, FormattedMessage } from 'react-intl';
+import { get } from 'lodash';
+import { Header } from '@buffetjs/custom';
+import { Envelope } from '@buffetjs/icons';
+import {
+  FormBloc,
+  request,
+  SettingsPageTitle,
+  SizedInput,
+  validateInput,
+} from 'strapi-helper-plugin';
+import getTrad from '../../utils/getTrad';
+import { AlignedButton, Text } from './components';
+
+const SettingsPage = () => {
+  const { formatMessage } = useIntl();
+  const [isTestButtonLoading, setIsTestButtonLoading] = useState(false);
+  const [showLoader, setShowLoader] = useState(false);
+  const [config, setConfig] = useState({
+    provider: '',
+    settings: { defaultFrom: '', defaultReplyTo: '', testAddress: '' },
+  });
+  const [providers, setProviders] = useState([]);
+  const [testAddress, setTestAddress] = useState();
+  const [testSuccess, setTestSuccess] = useState(false);
+
+  const title = formatMessage({ id: getTrad('Settings.title') });
+
+  const handleEmailTest = () => {
```

Comment:
I think it is better to keep the same logic in all our forms. The logic is the following, we use yup to validate our data and then if the data is correct we perform our request.

So, here you will need to make the following modifications:

1. Add the formError state
`./utils/schema.js` 

```
import * as yup from 'yup';
import { translatedErrors } from 'strapi-helper-plugin';

const schema = yup.object().shape({
  email: yup
    .string()
    .email(translatedErrors.email)
    .required(translatedErrors.required),
});

export default schema;
```

`index.js`

```
import schema from './utils/schema';

// ...

const [formErrors, setFormErrors] = useState({});
```

2. Add the isMounted logic
```
const isMounted = useRef(true);

useEffect(() => {
  return () => {
    isMounted.current = false;
  };
}, []);
```
3. modify the handler

```
// We just need to rename the function in order to stay consistent in the codebase.
const handleSubmit = async e => {
  e.preventDefault();
  let errors = {};

  try {
    await schema.validate({ email: testAddress }, { abortEarly: false });

    try {
      setIsTestButtonLoading(true);

      await request('/email/test', {
        method: 'POST',
        body: { to: testAddress },
      });

      setTestSuccess(true);

      strapi.notification.success(
        formatMessage({ id: getTrad('Settings.notification.test.success') }, { to: testAddress })
      );
    } catch (err) {
      strapi.notification.error(
        formatMessage({ id: getTrad('Settings.notification.test.error') }, { to: testAddress })
      );
    } finally {
      if (isMounted.current) {
        setIsTestButtonLoading(false);
      }
    }
  } catch (err) {
    errors = getYupInnerErrors(err);
    console.log(errors);
  }

  if (isMounted.current) {
    setFormErrors(errors);
  }
};
```